curvefs/src/client/inode_cache_manager.cpp

```diff
@@ -140,6 +164,55 @@ void InodeCacheManagerImpl::FlushInodeOnce() {
     }
 }
 
+void InodeCacheManagerImpl::AddParent(uint64_t inodeId, uint64_t parentId) {
+    curve::common::LockGuard lg2(parentIdMapMutex_);
+    if (parentIdMap_.count(inodeId)) {
+        parentIdMap_[inodeId].emplace_back(parentId);
```

Comment:
what if add same parentid more than one time?

Reply:
bool InodeCacheManagerImpl::UpdateParent(uint64_t inodeId, uint64_t oldParentId,
    uint64_t newParentId) {
    curve::common::LockGuard lg2(parentIdMapMutex_);
    if (parentIdMap_.count(inodeId)) {
        auto iter = std::find(parentIdMap_[inodeId].begin(),
            parentIdMap_[inodeId].end(), oldParentId);
        if (iter != parentIdMap_[inodeId].end()) {
            *iter = newParentId;
            return true;
        }
    }
    return false;
}

for example, if parent added twice, UpdateParent only update one, another is not change

Reply:
> 

That's ok, if make a hard link at the same dir wil have two same parent and rename one to another dir will only change one parent.